bsp4j/src/main/java/ch/epfl/scala/bsp4j/Protocol.xtend

```diff
@@ -724,3 +737,22 @@ class DependencyModule {
     this.version = version
   }
 }
+
+@JsonRpcData
+class FormatParams {
+  @NonNull List<BuildTargetIdentifier> targets
```

Comment:
Is scala-cli able to format a subset of files already?

Reply:
![image](https://user-images.githubusercontent.com/24596613/223998381-813e86a2-ee19-404d-b8fd-81cb31093843.png)

Yes it does - at least using CLI. In the screenshot above I've done the formatting of only one file in the directory


Another question is how much preciese show this option be. We can accept a list of files for each target for example - in that case the parameters would look like this:

```
[
  {
    "targetId": ...,
    "filesToFormat": [ // list of files and/or directories ]
  }
]
```

If list `filesToFormat` is empty - then the whole target should be formatted.

Another option is the following


```
[
  {
    "targetId": ...,
    "fileToFormat": "individual file (optional)"
  }
]
```

In this example we have only one optional parameter - if it passed then the passed file should be formatted, otherwise the whole target should be formatted. It looks more simple and it's useful in a particular case - to format the file you are currently working on in the editor/IDE

Reply:
I would choose the first option with an array of files because that's more flexible and probably not harder to implement.